### PR TITLE
Changes for new version control system

### PR DESCRIPTION
**Pull request by Janglee123**
It is based on libgit2-bump-fix and requires [vcs-new-ui](https://github.com/Janglee123/godot/tree/vcs-new-ui) to work. Both branches are updated parallelly.

### CODE REVIEW

**Comment by IronicallySerious** [comment]
@Janglee123 Could you rebase this on the current master? Also, there are some merge conflicts now, which I guess are quite small. Just use the build_libs.bat changes exactly as in the master branch, except if you are adding some new CMake flags

File: SConstruct [hunk 1]
```diff
@@ -87,7 +87,7 @@ cpp_library += '.' + str(bits)
 # make sure our binding library properly includes
 env.Append(CPPPATH=['.', godot_headers_path, cpp_bindings_path + 'include/', cpp_bindings_path + 'include/core/', cpp_bindings_path + 'include/gen/'])
 env.Append(LIBPATH=[cpp_bindings_path + 'bin/', libgit2_lib_path])
-env.Append(LIBS=[cpp_library, 'git2'])
+env.Append(LIBS=[cpp_library, 'ssl', 'crypto', 'ssh2', 'git2'])
```
Comment (IronicallySerious) [inline SConstruct:90]: Also this should be added in the x11 section above, as I said on Matrix

**Review by IronicallySerious — CHANGES_REQUESTED** [review]

**Comment by ChronicallySerious** [comment]
Simply updating this branch to master by a basic merge for now because all that has changed in master, which is minimal C++ code changes but still a lot of build scripts and CI configs.

**Comment by Janglee123** [comment]
Updated for Godot 3.x branch.

**Comment by ChronicallySerious** [comment]
Updating this branch to run macOS arm arch CI

**Pull request merged**